Content.Server/Body/Systems/TransferMindOnGibSystem.cs

```diff
@@ -0,0 +1,50 @@
+using Content.Server.Body.Components;
+using Content.Server.Mind.Components;
+
+namespace Content.Server.Body.Systems
+{
+    public sealed class TransferMindOnGibSystem : EntitySystem
+    {
+        [Dependency] private readonly SkeletonBodyManagerSystem _skeletonBodyManager = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<TransferMindOnGibComponent, BeforeGibbedEvent>(OnBeforeGibbed);
+            SubscribeLocalEvent<TransferMindOnGibComponent, BeingGibbedEvent>(OnBeingGibbed);
+        }
+
+        private void OnBeforeGibbed(EntityUid uid, TransferMindOnGibComponent component, BeforeGibbedEvent args)
+        {
+            if (TryComp<MindComponent>(uid, out var mindComp) && mindComp.Mind != null)
+            {
+                if (TryComp<BodyComponent>(uid, out var bodyComp))
+                {
+                    foreach (var part in bodyComp.Parts)
+                    {
+                        var entity = part.Key.Owner;
+                        if (HasComp<SkeletonBodyManagerComponent>(entity))
+                        {
+                            component.TransferTarget = entity;
+                            _skeletonBodyManager.UpdateDNAEntry(entity, uid);
+                            mindComp.Mind.TransferTo(entity);
+                        }
+                    }
+                }
+            }
```

Comment:
invert control flow here

```suggestion
            if (!TryComp<MindComponent>(uid, out var mindComp) || mindComp?.Mind == null)
                return;

            if (!TryComp<BodyComponent>(uid, out var bodyComp))
                return;

            foreach (var part in bodyComp.Parts)
            {
                var entity = part.Key.Owner;
                if (HasComp<SkeletonBodyManagerComponent>(entity))
                {
                    component.TransferTarget = entity;
                    _skeletonBodyManager.UpdateDNAEntry(entity, uid);
                    mindComp.Mind.TransferTo(entity);
                }
            }
```

this is just generally a convention throughout the codebase since highly-intended control flow is really garbage to read